pkg/controller/migplan/validation.go

```diff
@@ -759,6 +773,157 @@ func (r ReconcileMigPlan) validatePvSelections(plan *migapi.MigPlan) error {
 	return nil
 }
 
+// Validate proxy secrets. Should only exist 1 or none
+func (r ReconcileMigPlan) validateRegistryProxySecrets(plan *migapi.MigPlan) error {
+	err := r.validateSourceRegistryProxySecret(plan)
+	if err != nil {
+		return err
+	}
+	return r.validateDestinationRegistryProxySecret(plan)
+}
+
+// Validate registry proxy secret ensuring it has the proper keys
+func (r *ReconcileMigPlan) validateRegistryProxySecret(secret *kapi.Secret) bool {
+	fields := []string{
+		settings.HttpProxy,
+		settings.HttpsProxy,
+		settings.NoProxy,
+	}
+	for _, key := range fields {
+		if _, found := secret.Data[key]; found {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Validate source proxy secret
+func (r ReconcileMigPlan) validateSourceRegistryProxySecret(plan *migapi.MigPlan) error {
+	if plan.Status.HasAnyCondition(Suspended, InvalidSourceClusterRef, SourceClusterNotReady) {
+		return nil
+	}
+
+	list := kapi.SecretList{}
+	selector := k8sLabels.SelectorFromSet(map[string]string{
+		"migration-proxy-config": "true",
+	})
+
+	// Source cluster proxy secret validation
+	srcCluster, err := plan.GetSourceCluster(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+
+	if srcCluster == nil || !srcCluster.Status.IsReady() {
```

Comment:
The IsReady() here is fine but redundant with the HasAnyCondition() check on `SourceClusterNotReady`.